realm/realm-library/src/main/java/io/realm/DynamicRealmObject.java

```diff
@@ -905,12 +905,11 @@ public String toString() {
                 case OBJECT:
                     sb.append(proxyState.getRow$realm().isNullLink(columnIndex)
                             ? "null"
-                            : Table.tableNameToClassName(proxyState.getRow$realm().getTable().getLinkTarget(columnIndex).getName()));
+                            : proxyState.getRow$realm().getTable().getLinkTarget(columnIndex).getClassName());
                     break;
                 case LIST:
-                    final String tableName = proxyState.getRow$realm().getTable().getLinkTarget(columnIndex).getName();
-                    String targetType = Table.tableNameToClassName(tableName);
-                    sb.append(String.format("RealmList<%s>[%s]", targetType, proxyState.getRow$realm().getLinkList(columnIndex).size()));
+                    String tableClassName = proxyState.getRow$realm().getTable().getLinkTarget(columnIndex).getClassName();
```

Comment:
Yeah, point taken.  This is the `className`, not the `tableName`.  Unfortunately, there is, already, a variable named "className" in scope.  I thought `tableClassName` was better than what AS suggested: `className1`.